superset/dashboards/api.py

```diff
@@ -1046,14 +1046,14 @@ def trigger_celery() -> WerkzeugResponse:
             cache_dashboard_screenshot.delay(
                 username=get_current_user(),
                 guest_token=g.user.guest_token
-                if isinstance(g.user, GuestUser)
+                if hasattr(g, "user") and isinstance(g.user, GuestUser)
                 else None,
                 dashboard_id=dashboard.id,
                 dashboard_url=dashboard_url,
+                cache_key=cache_key,
                 force=True,
                 thumb_size=thumb_size,
                 window_size=window_size,
-                cache_key=cache_key,
```

Comment:
nit: Is there a reason we're reordering these?

Reply:
No not really, just readability